components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleCaption isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @11,
+        UIContentSizeCategorySmall : @11,
+        UIContentSizeCategoryMedium : @11,
+        UIContentSizeCategoryLarge : @12,
+        UIContentSizeCategoryExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraLarge : @16,
+        UIContentSizeCategoryExtraExtraExtraLarge : @18,
+        UIContentSizeCategoryAccessibilityMedium : @20,
+        UIContentSizeCategoryAccessibilityLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @26,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @28
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleOverline isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @8,
+        UIContentSizeCategorySmall : @8,
+        UIContentSizeCategoryMedium : @9,
+        UIContentSizeCategoryLarge : @10,
+        UIContentSizeCategoryExtraLarge : @12,
+        UIContentSizeCategoryExtraExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraExtraLarge : @16,
+        UIContentSizeCategoryAccessibilityMedium : @18,
+        UIContentSizeCategoryAccessibilityLarge : @20,
+        UIContentSizeCategoryAccessibilityExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @26
+      };
+      _scalingCurve = scalingCurve;
+    } else {
+      // If nothing matches, return the metrics for MDCTextStyleBody1
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @13,
+        UIContentSizeCategorySmall : @14,
+        UIContentSizeCategoryMedium : @15,
+        UIContentSizeCategoryLarge : @16,
+        UIContentSizeCategoryExtraLarge : @18,
+        UIContentSizeCategoryExtraExtraLarge : @20,
+        UIContentSizeCategoryExtraExtraExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityMedium : @26,
+        UIContentSizeCategoryAccessibilityLarge : @30,
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scalableFontWithFont:(UIFont *)font {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
```

Comment:
Should this ordering be flipped? Is there any reason why we prefer the application's contentSizeCategory instead of the main screen's?

Reply:
Screen is probably a better first choice.  I'll invert.  Moved to a private implementation since it is used by MDCFontScaler and UIFont+MaterialScalable.